BlueMapCore/src/main/java/de/bluecolored/bluemap/core/storage/s3/S3Storage.java

```diff
@@ -0,0 +1,342 @@
+            throw new IOException(e);
+        }
+    }
+
+    @Override
+    public Optional<MetaInfo> readMetaInfo(String mapId, String name) throws IOException {
+        String path = getMetaFilePath(mapId, name);
+        try {
+            var info = client.headObject(
+                    HeadObjectRequest
+                            .builder()
+                            .bucket(bucket)
+                            .key(path)
+                            .build()
+            );
+            return Optional.of(new MetaInfo() {
+                @Override
+                public InputStream readMeta() throws IOException {
+                    return S3Storage.this.readMeta(mapId, name)
+                            .orElseThrow(() -> new IOException("Meta no longer present!"));
+                }
+
+                @Override
+                public long getSize() {
+                    return info.contentLength();
+                }
+            });
+        } catch (Exception e) {
+            if (e instanceof NoSuchKeyException) {
+                return Optional.empty();
+            }
+            throw new IOException(e);
+        }
+    }
+
+    @Override
+    public void deleteMeta(String mapId, String name) throws IOException {
+        String path = getMetaFilePath(mapId, name);
+        try {
+            client.deleteObject(
+                    DeleteObjectRequest
+                            .builder()
+                            .bucket(bucket)
+                            .key(path)
+                            .build()
+            );
+        } catch (Exception e) {
+            if (e instanceof NoSuchKeyException) {
+                return;
+            }
+            throw new IOException(e);
+        }
+    }
+
+    @Override
+    public void purgeMap(String mapId, Function<ProgressInfo, Boolean> onProgress) throws IOException {
+        String directory = getFilePath(mapId);
+        try {
+            var files = client.listObjectsV2Paginator(
+                    ListObjectsV2Request
+                            .builder()
+                            .bucket(bucket)
+                            .prefix(directory + "/")
+                            .build()
+            );
+            var filesList = files.contents().stream().collect(Collectors.toList());
+            for (int i = 0; i < filesList.size(); i++) {
+                var file = filesList.get(i);
+                client.deleteObject(
+                        DeleteObjectRequest
+                                .builder()
+                                .bucket(bucket)
+                                .key(file.key())
+                                .build()
+                );
+
+                if (!onProgress.apply(
+                        new ProgressInfo((double) (i + 1) / filesList.size())
+                )) return;
+            }
+        } catch (Exception e) {
+            if (e instanceof NoSuchKeyException) {
+                return;
```

Comment:
Can we really ignore this here? Since it interrupts the deletion-process, it should either throw an error, or better catched earlier so it doesn't interrupt the rest of the deletions.

Reply:
yeah there should be a try-catch in the loop body there